### PR TITLE
feat: store incoming migrations in local DB

### PR DESCRIPTION
**Pull request by kevin-dp**

### CODE REVIEW

**Comment by alco** [comment]
@kevin-dp Looks like you added `clients/typescript/.idea` to Git by accident.

File: clients/typescript/migrate.sh
Comment (kevin-dp) [inline clients/typescript/migrate.sh:1]: Ignore this script.
It is outdated and will be replaced by the script in the web example app:
[https://github.com/electric-sql/electric/blob/1da282ed67780ad61c9ead1dfc57762f64aed5e5/examples/web-wa-sqlite/migrate.sh](https://github.com/electric-sql/electric/blob/1da282ed67780ad61c9ead1dfc57762f64aed5e5/examples/web-wa-sqlite/migrate.sh)

File: clients/typescript/src/migrators/index.ts [hunk 1]
```diff
@@ -1,6 +1,11 @@
+//import { Statement } from '../util'
+
+import { Statement } from '../util'
+
 export { BundleMigrator } from './bundle'
 export { MockMigrator } from './mock'
 
+/*
```
Comment (icehaunter) [inline clients/typescript/src/migrators/index.ts:8]: [That's a lot of comments!](https://www.youtube.com/watch?v=IRl3r4q_dUY)
Let's maybe delete them for a nicer diff?

**Review by icehaunter — CHANGES_REQUESTED** [review]
LGTM with three nitpicks

File: clients/typescript/src/satellite/client.ts [hunk 2]
```diff
@@ -732,14 +732,22 @@ export class SatelliteClient extends EventEmitter implements Client {
       }
 
       if (op.migrate) {
+        // store the version of this migration transaction
+        // (within 1 transaction, every SatOpMigrate message
+        //  has the same version number)
+        // TODO: in the protocol: move the `version` field to the SatOpBegin message
+        //       or replace the `is_migration` field by an optional `version` field
```
[the remark this answers is not in the document]
Comment (kevin-dp) [reply clients/typescript/src/satellite/client.ts:739]: Right, created a linear [issue](https://linear.app/electric-sql/issue/VAX-718/move-migration-version-field-to-satopbegin-in-protocol) for it and referenced the issue ID in this TODO in the code.

File: clients/typescript/src/migrators/index.ts (hunk 1 above)
Comment (kevin-dp) [reply clients/typescript/src/migrators/index.ts:8]: Woops, yeah left them there in case i had to revert but this can go now. Thanks for catching this

**Review by icehaunter — APPROVED** [review]

**Pull request merged**